rllib/algorithms/sac/sac.py

```diff
@@ -332,13 +351,53 @@ def validate(self) -> None:
             )
             try_import_tfp(error=True)
 
+        if (
+            self.uses_new_env_runners
+            and self.replay_buffer_config["type"] != "EpisodeReplayBuffer"
+        ):
+            raise ValueError(
+                "When using the new `EnvRunner API` the replay buffer must be of type "
+                "`EpisodeReplayBuffer`."
+            )
+
+        if self.uses_new_env_runners:
+            self.model.update({"initial_alpha": self.initial_alpha})
```

Comment:
This should not go here. We should never change the config inside `validate()` (it should only ever warn or error, but never taper with the contents of a AlgorithmConfig object).

Actually, let's use this opportunity to fix this problem. Let's brainstorm solutions together in a chat :) How do we get AlgorithmConfig properties/settings into the `model_config` dict in a user friendly manner?

Also, this parameter should probably go into the SACLearner, it has nothing to do with any RLModule forward pass and is only used in the loss.

Reply:
Yes, I think this will be a general matter as some other algorithms will need it.

Reply:
Yup it is in the  `SACLearner`.